test/northwind/repo_test.exs

```diff
@@ -148,6 +148,12 @@ defmodule Northwind.RepoTest do
     |> Repo.preload(shipper: :orders)
   end
 
+  test "Support json_extract_path expression" do
+    from(e in Model.Employee)
+    |> where([e], e.metadata["twitter"] == "@andrew_fuller")
+    |> Repo.one!()
+  end
```

Comment:
In https://hexdocs.pm/ecto/Ecto.Query.API.html#json_extract_path/2 there are multiple ways to address the fields:

- `json_extract_path(post.meta, ["author", "name"])`
- `post.meta["author"]["name"]`
- `post.meta["tags"][0]["name"]`
- `post.meta["author"][^field]`

Best if each enumerated style is covered in a separate stanza. This way we are sure that everything evaluates to the same thing.

Reply:
I've included tests for the above variations.